primitives/src/lib.rs

```diff
@@ -37,6 +37,36 @@ impl TruncateFixedPointToInt for UnsignedFixedPoint {
     }
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct VaultCurrencyPair<CurrencyId: Copy> {
```

Comment:
Yea this is a leftover of back when I mistakenly thought we would be indexing stuff in staking/reward on the currencypair. We will actually have to do indexing on the currencypair eventually - e.g. we will need to have thresholds and liquidation vaults per currencypair. In that case it will be convenient, and it will prevent accidentally swapping the two currencies when passing them as arguments. I don't feel very strongly about it though, so let me know if you still want to flatten the structure given the above

Reply:
Thanks for this explanation, I agree it makes sense to keep it in that case!